src/components/Grid/index.tsx

```diff
@@ -3,20 +3,27 @@ import React from 'react'
 import { data } from '../../data/data'
 import { Card, Celebrity } from '../Card'
 import { Controls } from '../Controls'
+import SearchInput from '../SearchInput'
 
 export const Grid = () => {
   const [level, setLevel] = React.useState<number>(6)
   const [visible, setVisible] = React.useState<boolean>(true)
+  const [searchText, setSearchText] = React.useState('')
   const { celebrities } = data
 
   const renderBoxes = () =>
-    celebrities.map(celebrity => (
-      <Card
-        celebrity={celebrity as Celebrity}
-        level={level}
-        visible={visible}
-      />
-    ))
+    celebrities.map(celebrity => {
+      if (!celebrity.name.toLowerCase().includes(searchText.toLowerCase()))
```

Comment:
IMO, a bit hard to follow what this bit of code is trying to do. 
Maybe pull the logic out into a variable: 
```
const hasSearchText = searchText !== '' && celebrity.name.toLowerCase().includes(searchText.toLowerCase())
```
and in your if
```
if(!hasSearchText) {
    return null;
} 
```
Note that `searchText !== '' && ` bit helps with readability and follows a mindset of being specific. This will help maintaining the code in future.
Essentially you do the `includes` check only when user has started searching

Reply:
tnx, I used a variable to improve readability but I dont agree with `searchText !== '' &&` part because in that way if the searched text be empty, no item will be visible.